File: src/main/java/sk/hudak/prco/utils/ThreadUtils.java
package sk.hudak.prco.utils;

import lombok.extern.slf4j.Slf4j;

import java.util.Random;

@Slf4j
public final class ThreadUtils {

    private ThreadUtils() {
    }

    /**
     * Will sleep between 5 to 20 second
     */
    public static void sleepRandomSafe() {
        sleepRandomSafeBetween(5, 20);
    }

    public static void sleepRandomSafeBetween(int min, int max) {
        sleepSafe(generateRandomSecondInInterval(min, max));
    }

    public static int generateRandomSecondInInterval() {
        return generateRandomSecondInInterval(5, 20);
    }


    public static int generateRandomSecondInInterval(int min, int max) {
        return new Random().nextInt(max - min + 1) + min;
    }

    public static void sleepSafe(int second) {
        try {
            log.debug("start sleeping for {} sec", second);
            Thread.sleep(1000l * second);

        } catch (InterruptedException e) {
            //FIXME log
            e.printStackTrace();
        }
    }
}

File: src/main/java/sk/hudak/prco/utils/ThreadUtils.kt
package sk.hudak.prco.utils

import org.slf4j.LoggerFactory
import java.util.*


object ThreadUtils {

    val log = LoggerFactory.getLogger(ThreadUtils::class.java)

    /**
     * Will sleep between 5 to 20 second
     */
    @JvmStatic
    fun sleepRandomSafe() {
        sleepRandomSafeBetween(5, 20)
    }

    @JvmStatic
    fun sleepRandomSafeBetween(min: Int, max: Int) {
        val second = generateRandomSecondInInterval(min, max)
        sleepSafe(second)
    }

    @JvmStatic
    @JvmOverloads
    fun generateRandomSecondInInterval(min: Int = 5, max: Int = 20): Int {
        return Random().nextInt(max - min + 1) + min
    }

    @JvmStatic
    fun sleepSafe(second: Int) {
        try {
            log.debug("start sleeping for $second sec")
            Thread.sleep(1000L * second)

        } catch (e: InterruptedException) {
            //FIXME log
            e.printStackTrace()
        }

    }
}
